### PR TITLE
feature(outcomes): Track event sizes in outcomes

### PR DESCRIPTION
**Pull request by nathan-heskia**
Adds tsdb models for tracking the size in bytes of outcomes and enables querying for it at the organization or project level.

### CODE REVIEW

File: src/sentry/utils/outcomes.py [hunk 1]
```diff
@@ -119,6 +137,7 @@ def track_outcome(org_id, project_id, key_id, outcome, reason=None, timestamp=No
     assert isinstance(key_id, (type(None), six.integer_types))
     assert isinstance(outcome, Outcome)
     assert isinstance(timestamp, (type(None), datetime))
+    assert isinstance(event_size, (type(None), six.integer_types))
```
Comment (fpacifici) [inline src/sentry/utils/outcomes.py]: Maybe unrelated with this PR, but looking at the size method on the Event class, it is the size in characters of the json serialization of the data field of the event. That may be different than the size in bytes and it is the size of an event post normalization and can be tweaked during event save. 
IS this the expected way to measure event size ?

**Review by fpacifici — COMMENTED** [review]
Seems generally ok to me. One clarification and did you do one end to end test with Snuba? If not could you try to ensure everything is sane ?

File: src/sentry/utils/outcomes.py (hunk 1 above)
Comment (nathan-heskia) [reply src/sentry/utils/outcomes.py]: I think there is an open question about what constitutes an event size. The goal of this pull request is to add this metric to the outcome, knowing that the implementation of event size for billing purposes is still to come.

File: src/sentry/tsdb/snuba.py [hunk 2]
```diff
@@ -85,6 +85,24 @@ class SnubaTSDB(BaseTSDB):
             "times_seen",
             [["outcome", "=", outcomes.Outcome.FILTERED]],
         ),
+        TSDBModel.organization_total_bytes_received: SnubaModelQuerySettings(
+            snuba.Dataset.Outcomes,
+            "org_id",
+            "bytes_received",
```
Comment (fpacifici) [inline src/sentry/tsdb/snuba.py]: I am not sure I get where we use sum instead of count as aggregation function for these cases when querying snuba.

**Review by fpacifici — CHANGES_REQUESTED** [review]

**Pull request closed without merging**